terms/hyperparameters_nm.tex

```diff
@@ -0,0 +1,70 @@
+
+\subsection{Bandit-Based Approach in Conjunction with Adaptive Allocation}
+ The HYPERBAND system was constructed by the same authors as above. It attempts to evaluate randomly-sampled hyperparameter configurations by adaptively assigning the corresponding amount of resources according to validation losses. This is termed a bandit-based approach since the project makes use of 'infinite armed bandits' in its implementation. Bandit arms represent mean-level reward distributions from particular experiments. The reward budget of each arm is essential for selection of the best distributions.  As detailed by Wang \textit{et al.} in \cite{Wang2009}, multi-armed bandit problems focus on \textit{exploration} vs. \textit{exploitation}. Arms can be `exploited', meaning that they performed well and may be selected for their respective task. Exploration may entail pulling entirely new arms, or pulling already discovered ones of interest. Scoping back to the HYPERBAND paper, the focus is on exploration infinite-armed bandit problems, selecting arms within the top reward percentiles for hyperparameter optimization.
+
+
+\section{Scenario: Hyperparameter Tuning for Gradient Descent}
+This scenario will look into a practical example of how hyperparameter values affect the performance of a gradient descent algorithm. Consider the following setup with cost function:
+
+\begin{eqnarray}
+y = 4x^{2} - 2x - 1 \\
+dy/{dx} = 8x-2
+\end{eqnarray}
+
+\noindent The objective of gradient descent algorithms is to minimize a function. Mathematically, this implies that the algorithm will cycle through different values of $x$ until a minimum value of y is obtained, generally to a predefined precision. This process may be represented by the decrease in magnitude of the gradient of the function, hence the inclusion of the first derivative. The primary hyperparameter to be optimized in this case will be $\alpha$, representing the learning rate. As previously stated, the learning rate will affect the capability of the system to evolve. Values are calculated as follows for each iteration:
+
+\begin{eqnarray}
+x_{new} = x_{i} - \alpha(dy/{dx_{i}}) \\
+y_{new} = (4(x_{new}^{2}) - 2x_{new} - 1  
+\end{eqnarray}
+
+\noindent  Another important parameter in this setup is $x_{0}$, which is the initial x-value to be used as an estimate for the minimum point. The following plot shows the algorithm with an $\alpha$ of $0.0001$:
+
+ \begin{figure}
+	\begin{center}
+		\centering{{\includegraphics[width=0.82\textwidth]{graphics/hyperparameters/Grad_Desc_LR0001}}}
```

Comment:
No need for graphics in the path (its in the search part in the main document)
Also, refer to figures as Figure 4 etc. not Figure above ... you have no guarantee latex will place it above, on same page etc.

Same for all other figures/images